messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/bridge/JMSBridgeHandler.java

```diff
@@ -47,101 +45,110 @@
+            if (!readOnly) {
+                throw ControllerLogger.ROOT_LOGGER.managementResourceNotFound(context.getCurrentAddress());
             }
+            return;
         }
-        else if (START.equals(operationName)) {
-            try {
-                // we do not start the bridge directly but call startBridge() instead
-                // to ensure the class loader will be able to load any external resources
-                JMSBridgeService service = (JMSBridgeService) bridgeService.getService();
-                service.startBridge();
-            } catch (Exception e) {
-                context.getFailureDescription().set(e.getLocalizedMessage());
-            }
-        } else if (STOP.equals(operationName)) {
-            try {
-                bridge.stop();
-            } catch (Exception e) {
-                context.getFailureDescription().set(e.getLocalizedMessage());
-            }
-        } else if (PAUSE.equals(operationName)) {
-            try {
-                bridge.pause();
-            } catch (Exception e) {
-                context.getFailureDescription().set(e.getLocalizedMessage());
-            }
-        } else if (RESUME.equals(operationName)) {
-            try {
-                bridge.resume();
-            } catch (Exception e) {
-                context.getFailureDescription().set(e.getLocalizedMessage());
-            }
-        } else {
-            throw MessagingLogger.ROOT_LOGGER.unsupportedOperation(operationName);
+        final JMSBridge bridge = JMSBridge.class.cast(bridgeService.getValue());
+        switch (operationName) {
+            case READ_ATTRIBUTE_OPERATION:
+                readAttributeValidator.validate(operation);
+                final String name = operation.require(NAME).asString();
+                if (STARTED.equals(name)) {
+                    context.getResult().set(bridge.isStarted());
+                } else if (PAUSED.getName().equals(name)) {
+                    context.getResult().set(bridge.isPaused());
+                }  else if (CommonAttributes.MESSAGE_COUNT.getName().equals(name)) {
+                    context.getResult().set(bridge.getMessageCount());
+                }  else if (ABORTED_MESSAGE_COUNT.getName().equals(name)) {
+                    context.getResult().set(bridge.getAbortedMessageCount());
+                } else {
+                    throw MessagingLogger.ROOT_LOGGER.unsupportedAttribute(name);
+                }
+                break;
+            case START:
+                try {
+                    // we do not start the bridge directly but call startBridge() instead
+                    // to ensure the class loader will be able to load any external resources
+                    JMSBridgeService service = (JMSBridgeService) bridgeService.getService();
+                    service.startBridge();
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
```

Comment:
I think an `OperationFailedException` might be better here. With a `RuntimeException` the stack trace will always be logged which is maybe what we're after here. Previously though this just set a failure description.

Reply:
@jamezp An OFE should only be used to indicate a caller mistake, not a server fault.

Only setting a failure description means a server fault occurred but there would be nothing in the server log to indicate that.

This handling is good, other than perhaps catching and throwing on any RuntimeException and only wrapping checked exceptions. But that's a nit.

Reply:
An OFE would be appropriate if the bridge was already started and that's what caused startBridge to throw an exception. Looking at the code though if that happened there'd be no exception, just an error msg in the server log.

Reply:
@jamezp BTW the reason this handling changed vs previously in response to my review comment on this file from 10 days ago. :)

Reply:
Perfect thanks for the explanation!